rotkehlchen/externalapis/etherscan.py

```diff
@@ -238,7 +241,16 @@ def _query(
         an unexpected response is returned. Also in the case of exhausting the backoff time.
         """
         result = None
-        query_str = f'https://{self.prefix_url}{self.base_url}/api?module={module}&action={action}'
+        if self.cached_settings.get_settings().use_unified_etherscan_api:
```

Comment:
I don't get what this is. I mean I get the sort of general idea but the details are very hard to understand from the code and very obscure.

Waht's more it adds multiple if conditions to each and every etherscan api call.

That's a no-go. A setting like that should just setup etherscan queries in a specific way but not add multiple conditionals to every call

Reply:
Changed this to use an approach like what was suggested in [discord](https://discord.com/channels/657906918408585217/1080783409640648765/1322179131550732370), which removes the if conditions from the query function.